### PR TITLE
Fix ClassValidationPipe

### PR DESCRIPTION
**Pull request by notaphplover**
### Changed
- `undefined` input is allowed on undefined parameters. `undefined` inputs no longer trigger non inversify errors.
- Nullish values are allowed on nullable types.
- Non plain object types are validated unless some exceptions like Buffer.

### Context
- See #1506.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of missing or null/undefined request payloads: returns clear validation errors for required bodies and treats nullish/primitive inputs more predictably.

* **Tests**
  * Expanded coverage for undefined, null, string, number, and object inputs and related validation edge cases.

* **Refactor**
  * Reorganized and standardized test suites and mock initialization for better maintainability.

* **Chores**
  * Added a changeset entry for a patch release.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/framework/validation/libraries/class-validator/src/pipes/ClassValidationPipe.ts [hunk 1]
```diff
@@ -7,6 +7,25 @@ import {
...
+  Array,
+  BigInt,
+  Boolean,
+  Buffer,
+  Function,
+  Number,
+  Object,
+  String,
+  Symbol,
+];
```
[the remark this answers is not in the document]
Comment (notaphplover) [reply packages/framework/validation/libraries/class-validator/src/pipes/ClassValidationPipe.ts:27]: This is a NodeJS based library, Pipes is strongly coupled to the http framework
